### ISSUE TITLE
training time isomer?

### ISSUE THREAD

**Issue by zht8506** [DLUT-yyc/Isomer#1]
May I ask how long you need to train isomer. It takes 1 hour for me to train one epoch, and 500h for 500 epochs, which is too long

**Comment by DLUT-yyc** [comment]
Hi,

Due to the use of Transformer architecture, Isomer does require a long training time. 

Here are some strategies I suggest to expedite the process:

(1) In my ablation studies, I typically utilize only a subset of data from YoutubeVOS (for instance, sampling one every ten frames) to hasten the experimental progression during the pre-training phase. Once the model's specifics and hyperparameters are established, I proceed to train it on the complete dataset to ensure a fair comparison with other methods (e.g., HFAN).

(2) My code incorporates parallel training (ddp), enabling the use of multiple GPUs to accelerate the experiment. However, in my experience, training with a single GPU often yields a little better results compared to using multiple GPUs.

(3) The optimal batch size I've adjusted on my machine is 8 per device, only consuming approximately 6GB of GPU memory. If you have more powerful GPUs, you might consider increasing the batch size.

(4) Fine-tuning with the FBMS dataset may harm the performance on the DAVIS-16 validation set. As such, it might be beneficial to consider using only the DAVIS training set for the fine-tuning process.

Good luck!

**Comment by zht8506** [comment]
Thank you very much for your detailed reply and, have a nice day.

**Comment by zht8506** [comment]
I would like to ask, where can I modify the code to achieve sampling 1 frame every 30 frames? thank you very much

**Comment by DLUT-yyc** [comment]
Hi,

This part of the implementation is not present in the code I released. I just use a simple script to sample the images of youtubevos in a loop and store them in another folder. 

The following functions may help you:

os.listdir()
os.path.join()
os.system()
os.makedirs()

Good luck!

**Comment by zht8506** [comment]
Thank you for your reply. I get it. Wish you all the best.

**Comment by TianXuXiang** [comment]
> Hi,
> 
> This part of the implementation is not present in the code I released. I just use a simple script to sample the images of youtubevos in a loop and store them in another folder.
> 
> The following functions may help you:
> 
> os.listdir() os.path.join() os.system() os.makedirs()
> 
> Good luck!

你好，可以加个微信讨论一下吗
